src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -34,25 +46,98 @@
     private static final String STUDY_SERVER_API_VERSION = "v1";
     private static final String ROOT_CATEGORY_REACTOR = "reactor.";
 
-    private WebClient webClient;
+    private final WebClient webClient;
     private String studyServerBaseUri;
 
     private final DirectoryElementRepository directoryElementRepository;
 
+    private static final String CATEGORY_BROKER_OUTPUT = DirectoryService.class.getName() + ".output-broker-messages";
+    private static final String CATEGORY_BROKER_INPUT = DirectoryService.class.getName() + ".input-broker-messages";
+
+    private static final Logger MESSAGE_OUTPUT_LOGGER = LoggerFactory.getLogger(CATEGORY_BROKER_OUTPUT);
+
+    static final String HEADER_USER_ID = "userId";
+    static final String HEADER_UPDATE_TYPE = "updateType";
+    static final String UPDATE_TYPE_DIRECTORIES = "directories";
+    static final String HEADER_DIRECTORY_UUID = "directoryUuid";
+    static final String HEADER_IS_PUBLIC_DIRECTORY = "isPublicDirectory";
+    static final String HEADER_IS_ROOT_DIRECTORY = "isRootDirectory";
+    static final String HEADER_ERROR = "error";
+    static final String HEADER_STUDY_UUID = "studyUuid";
+
+    private final StreamBridge studyUpdatePublisher;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DirectoryService.class);
+
     public DirectoryService(
             DirectoryElementRepository directoryElementRepository,
             @Value("${backing-services.study-server.base-uri:http://study-server/}") String studyServerBaseUri,
-            WebClient.Builder webClientBuilder) {
+            WebClient.Builder webClientBuilder, StreamBridge studyUpdatePublisher) {
         this.directoryElementRepository = directoryElementRepository;
         this.studyServerBaseUri = studyServerBaseUri;
 
         this.webClient = webClientBuilder.build();
+        this.studyUpdatePublisher = studyUpdatePublisher;
+    }
+
+    /* notifications */
+    @Bean
+    public Consumer<Flux<Message<String>>> consumeStudyUpdate() {
+        return f -> f.log(CATEGORY_BROKER_INPUT, Level.FINE).flatMap(message -> {
+            UUID studyUuid = UUID.fromString(message.getHeaders().get(HEADER_STUDY_UUID, String.class));
+            String error = message.getHeaders().get(HEADER_ERROR, String.class);
+            String userId = message.getHeaders().get(HEADER_USER_ID, String.class);
+            UUID parentUuid = getParentUuid(studyUuid);
+            if (error != null) {
+                deleteElement(studyUuid, userId).subscribe();
```

Comment:
Are we sure we want to do a deleteElement every time we receive a studyUpdate with an error ?